circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/AmazonS3ClientFactory.java

```diff
@@ -28,6 +27,7 @@
    * @param s3s3CopierOptions Copier options.
    * @return a new instance of {@linkplain AmazonS3 AmazonS3}
    */
-  AmazonS3 newInstance(AmazonS3URI uri, S3S3CopierOptions s3s3CopierOptions);
+  AmazonS3 newSourceInstance(AmazonS3URI uri, S3S3CopierOptions s3s3CopierOptions);
 
+  AmazonS3 newTargetInstance(AmazonS3URI uri, S3S3CopierOptions s3s3CopierOptions);
```

Comment:
Add javadoc here too plus a one liner description above the params that outlines what they do (mainly the difference between "source" and "target" factories).